src/main/java/usecase/impactor/MoneyImpactor.java

```diff
@@ -8,12 +8,27 @@
 public class MoneyImpactor {
 
     /**
-     * Impact the money of a player
-     * @param player player whose money should be changed
-     * @param value the value of money should be change. If it is a deduction, the value is negative
+     * change the money of two players
+     * @param price the amount of money to change
+     * @param giver the player to give money
      */
-    public void trading(Player player, int value){
-        int now = player.getMoney();
-        player.setMoney(now + value);
+    public void trade(int price, Player giver){
+        giver.setMoney(giver.getMoney() - price);
+
+        if (giver.getMoney() < 0) {
+            StatusImpactor statusImpactor = new StatusImpactor();
+            statusImpactor.changeStatus(giver, "playable", -8964);
+        }
+    }
+
+    /**
+     * change the money of two players
+     * @param price the amount of money to change
+     * @param giver the player to give money
+     * @param receiver the player to receive money
+     */
+    public void trade(int price, Player giver, Player receiver) {
+        this.trade(price, giver);
```

Comment:
if modify the previous function, i think this line should be `this.trade(-price, giver)`